src/utils/url.js

```diff
@@ -92,64 +94,32 @@ export const tryParseUrl = (url: string, base?: string | URL): URL | void => {
 };
 
 /**
- * Test if the given URL string resolves on the realm, with realm as base.
- *
- * DEPRECATED because URL strings are complicated.  Callers should construct
- *   a URL object before this point, and use the same URL object both for
- *   asking any questions like this one and for subsequently making any
- *   network requests.
- *
- * This returns true just if `new URL(url, realm)` gives a URL that is
- * within the same Zulip realm.
- *
- * This performs a call to `new URL` and therefore may take a fraction of a
- * millisecond.  Avoid using in a context where it might be called more than
- * 10 or 100 times per user action.
```

Comment:
Makes sense; thanks for pointing this out.

> (No need to actually go and memoize the call in Lightbox -- I see that by the end of the branch it's getting the URL as a prop.)

I've actually gone ahead and done this for the next revision. I think there's some chance we'll postpone (or merge and end up reverting) that change at the end of the branch, the one where the lightbox code gets a URL object instead of a string.

Do we know if #4136 also breaks the lightbox, or only non-image uploads? If it affects the lightbox, see https://github.com/zulip/zulip-mobile/issues/5518#issuecomment-1294515953, where I found a reason it might be not-so-good to be working with URL objects…🤯

Reply:
Oh OK great, yay!